token-locker/src/whitelist.rs

```diff
@@ -0,0 +1,50 @@
+use near_sdk::{assert_self, env, AccountId};
+
+use crate::*;
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum WhitelistMode {
+    Blocked,
+    CheckToken,
+    CheckAccountAndToken,
+}
+
+impl Contract {
+    pub fn set_token_whitelist_mode(&mut self, token: AccountId, mode: WhitelistMode) {
+        assert_self();
+        self.whitelist_tokens.insert(&token, &mode);
+    }
+
+    pub fn add_account_to_whitelist(&mut self, token: AccountId, account: AccountId) {
+        assert_self();
+        assert!(
+            self.whitelist_tokens.get(&token).is_some(),
+            "The token mode should be set first"
+        );
+        let token_account_key = format!("{}:{}", token, account);
```

Comment:
Please move `token_account_key` calculation to a separate function as it's used twice

Reply:
Fixed 3c8415c85f4f85a003ca9ee1ff925f2611807a09